test/e2e/scenarios/autorepair.go

```diff
@@ -0,0 +1,136 @@
+// +build e2e
+
+package scenarios
+
+import (
+	"context"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	. "github.com/onsi/gomega"
+	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
+	cmdcluster "github.com/openshift/hypershift/cmd/cluster"
+	awsutil "github.com/openshift/hypershift/cmd/infra/aws/util"
+	e2eutil "github.com/openshift/hypershift/test/e2e/util"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type TestAutoRepairOptions struct {
+	AWSCredentialsFile string
+	AWSRegion          string
+	PullSecretFile     string
+	ReleaseImage       string
+	ArtifactDir        string
+	BaseDomain         string
+}
+
+func TestAutoRepair(ctx context.Context, o TestAutoRepairOptions) func(t *testing.T) {
+	return func(t *testing.T) {
+		t.Parallel()
+		g := NewWithT(t)
+
+		client := e2eutil.GetClientOrDie()
+
+		// Create a namespace in which to place hostedclusters
+		namespace := e2eutil.GenerateNamespace(t, ctx, client, "e2e-clusters-")
+		name := e2eutil.SimpleNameGenerator.GenerateName("example-")
+
+		// Define the cluster we'll be testing
+		hostedCluster := &hyperv1.HostedCluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: namespace.Name,
+				Name:      name,
+			},
+		}
+
+		// Ensure we clean up after the test
+		defer func() {
+			// TODO: Figure out why this is slow
+			//e2eutil.DumpGuestCluster(context.Background(), client, hostedCluster, o.ArtifactDir)
+			e2eutil.DumpAndDestroyHostedCluster(t, context.Background(), hostedCluster, o.AWSCredentialsFile, o.AWSRegion, o.BaseDomain, o.ArtifactDir)
+			e2eutil.DeleteNamespace(t, context.Background(), client, namespace.Name)
+		}()
+
+		// Create the cluster
+		createClusterOpts := cmdcluster.Options{
+			Namespace:          hostedCluster.Namespace,
+			Name:               hostedCluster.Name,
+			InfraID:            hostedCluster.Name,
+			ReleaseImage:       o.ReleaseImage,
+			PullSecretFile:     o.PullSecretFile,
+			AWSCredentialsFile: o.AWSCredentialsFile,
+			Region:             o.AWSRegion,
+			// TODO: generate a key on the fly
+			SSHKeyFile:       "",
+			NodePoolReplicas: 3,
+			InstanceType:     "m4.large",
+			BaseDomain:       o.BaseDomain,
+			NetworkType:      string(hyperv1.OpenShiftSDN),
+			AutoRepair:       true,
+		}
+		t.Logf("Creating a new cluster. Options: %v", createClusterOpts)
+		err := cmdcluster.CreateCluster(ctx, createClusterOpts)
+		g.Expect(err).NotTo(HaveOccurred(), "failed to create cluster")
+
+		// Get the newly created cluster
+		err = client.Get(ctx, crclient.ObjectKeyFromObject(hostedCluster), hostedCluster)
+		g.Expect(err).NotTo(HaveOccurred(), "failed to get hostedcluster")
+		t.Logf("Found the new hostedcluster. Namespace: %s, name: %s", hostedCluster.Namespace, name)
```

Comment:
nit: can use `%s, client.ObjectKeyFromObject(hostedCluster)` which automatically calls `.String()` and prints `namespace/name`